src/bun.js/api/server.zig

```diff
@@ -4948,15 +4899,28 @@ pub fn NewServer(comptime ssl_enabled_: bool, comptime debug_mode_: bool) type {
             var ctx = this.request_pool_allocator.create(RequestContext) catch @panic("ran out of memory");
             ctx.create(this, req, resp);
             var request_object = this.allocator.create(JSC.WebCore.Request) catch unreachable;
+            var body = this.allocator.create(JSC.WebCore.Body.Value) catch unreachable;
+            // default body state is .Null
+            body.* = .{
+                .Null = {},
+            };
+
+            ctx.request_body = body;
+            const js_signal = JSC.WebCore.AbortSignal.create(this.globalThis);
+            js_signal.ensureStillAlive();
+            if (JSC.WebCore.AbortSignal.fromJS(js_signal)) |signal| {
+                ctx.signal = signal.ref().ref(); // +2 refs 1 for the request and 1 for the request context
+            }
 
             request_object.* = .{
                 .url = "",
                 .method = ctx.method,
                 .uws_request = req,
                 .https = ssl_enabled,
-                .body = .{
-                    .Null = {},
-                },
+                .signal = ctx.signal,
+                // ctx owns the body because ctx always live longer than request_object
```

Comment:
This isn't true. `Request` can outlive the `RequestContext` if the `Request` object escapes the `fetch` handler (like `globalThis.request = request`)

Reply:
Fixed here https://github.com/oven-sh/bun/pull/2752/commits/ad14e57e1bc6cc52082bd4798e3cacdbe92cf76e